src/main/java/com/ocelotslovebirds/birdhaus/blocks/BirdhouseBlockEntity.java

```diff
@@ -70,14 +84,41 @@ public void load(CompoundTag tag) {
      */
 
     public void tickServer() {
+        // Find all parrots in the vicinity
+        List<Parrot> parrots = this.level.getEntitiesOfClass(Parrot.class, this.birdHouseBB);
         if (counter > 0) {
             counter--;
             setChanged();
+
+            // Add the new goal to the parrots if they dont already have it
+            for (Parrot temp:parrots) {
+                // Check to see if they already have the goal applied. If not, apply it.
+                if (!temp.goalSelector.getAvailableGoals().stream()
+                    .anyMatch(wg -> wg.getGoal() instanceof HangAroundBirdhouseGoal)) {
+                    temp.goalSelector.addGoal(0, new HangAroundBirdhouseGoal(temp, 1.0, 60, false, this.getBlockPos()));
+                }
+            }
         }
         if (counter <= 0) {
             if (!itemHandler.extractItem(0, 1, false).isEmpty()) {
                 counter = 100;
                 setChanged();
+            } else {
+                for (Parrot temp:parrots) {
+                    // Check to see if they already have the goal applied. If not, apply it.
```

Comment:
why not `for (Parrot parrot:parrots) {`? Is there a reason why you went with `temp`? The comment here is inaccurate too I believe, you are removing the goal I think?

Reply:
Can the lingering duration be moved to the birdhouse goal? i.e. something like invalidating the goal after a certain amount of time has passed. I havent exactly tried running the mod yet with your changes, but wouldn't this mean that all the birds in the vicinity will flock over at the same time and leave simultaneously?

Reply:
Giving the birdhouse the role of removing the flocking goal seems to be potential source of future issues, e.g. if two birdhouses are side by side, birdhouse A might remove the goal set by birdhouse B

Reply:
For your second comment, they dont really act as soon as the goal is removed. Internally, parrots have a couple of different goals, so instead of immediately reacting to the goal being removed, whenever their ai step occurs, it just wont be a goal that they could access. They then slowly leave the area at different times using Parrot.parrotWanderGoal which allows them to wander to a nearby block in their vicinity. I dont think the lingering is really necessary because of this.

For the third comment, youre definitely right. Though as of right now, I can't immediately think of a way for the goal to be removed by logic coming from the parrot entity itself. I think maybe i can try figuring out how to properly remove the goal after the first milestone presentation.

Reply:
okay cool! I do have this very silly idea tho, the `canUse` method of the goal should return false after some period of time, as in if the goal has a time limit, then once we've passed the limit the goal should always return false. And then perhaps we can register some kind of global event to look for HangAroundBirdhouseGoal that are invalid and remove them every now and then. We don't have to do it now, but it is just an idea I wanted to put out there

Reply:
That's not a bad idea. There was another way that we could do it too, which acts a lot like the Villager POI system that might not be a bad idea either. We can talk about both ways of going about it for milestone 2.